src/libraries/System.Transactions.Local/src/System/Transactions/DtcProxyShim/DtcProxyShimFactory.cs

```diff
@@ -350,7 +352,13 @@ internal ITransactionTransmitter GetCachedTransmitter(ITransaction transaction)
     }
 
     internal void ReturnCachedTransmitter(ITransactionTransmitter transmitter)
-        => _cachedTransmitters.Enqueue(transmitter);
+    {
+        if (_cachedTransmitters.Count < _maxCachedInterfaces)
+        {
+            transmitter.Reset();
+            _cachedTransmitters.Enqueue(transmitter);
```

Comment:
Unless there's higher-level synchronization I don't see, races could result in going above the maximum.  Is that ok?

Reply:
Yeah, it should be fine. As above, all this is is a best-effort caching mechanism to recycle ITransactionReceivers/ITransactionTransmitters. If there end up being over-maximum instances in the queue, that's fine.

Reply:
Ok, cool.  Might be worth a comment for a future reader that the max limit is soft.

Reply:
Oops, didn't notice this before clicking merge. Will add a comment in #76376.